Sources/Public/Animation/LottieAnimationView.swift

```diff
@@ -648,15 +648,54 @@ open class LottieAnimationView: LottieAnimationViewBase {
     from dotLottieFile: DotLottieFile)
   {
     guard let dotLottieAnimation = dotLottieFile.animation(for: animationId) else { return }
+    loadAnimation(dotLottieAnimation, from: dotLottieFile)
+  }
 
-    loopMode = dotLottieAnimation.configuration.loopMode
-    animationSpeed = CGFloat(dotLottieAnimation.configuration.speed)
-
-    if let imageProvider = dotLottieAnimation.configuration.imageProvider {
-      self.imageProvider = imageProvider
-    }
+  /// Sets the lottie file backing the animation view. Setting this will clear the
+  /// view's contents, completion blocks and current state. The new animation will
+  /// be loaded up and set to the beginning of its timeline.
+  /// The loopMode, animationSpeed and imageProvider will be set according
+  /// to lottie file settings
+  /// - Parameters:
+  ///   - animationIndex: Internal animation index to play. Optional
+  ///   Defaults to play first animation in file.
+  ///   - dotLottieFile: Lottie file to play
+  public func loadAnimation(
+    animationIndex: Int,
+    from dotLottieFile: DotLottieFile)
+  {
+    guard let dotLottieAnimation = dotLottieFile.animation(for: animationIndex) else { return }
+    loadAnimation(dotLottieAnimation, from: dotLottieFile)
+  }
 
-    animation = dotLottieAnimation.animation
+  /// Exclusive for DotLottie animations. This will be ignored if playing json animation.
+  /// Plays the next animation in the array.
+  /// If there are no next animations, will loop back to the first animation in the array.
+  /// If there is only one animation, will reset to the beginning of the current animation.
+  /// Setting this will clear the view's contents, completion blocks and current state.
+  /// The new animation will be loaded up and set to the beginning of its timeline.
+  /// The loopMode, animationSpeed and imageProvider will be set according
+  /// to lottie file settings
+  public func nextAnimation() {
```

Comment:
I'm not sure that `nextAnimation()` and `previousAnimation()` need to be public APIs on `LottieAnimationView`. Before adding new public APIs I think we need to demonstrate a clear bar of:
 1. usefulness / how often this is needed
 2. difficulty or impracticality of implementing the functionality yourself in client code

Currently I'm thinking:
 1. I think this is totally feasible to implement in client code that uses Lottie, assuming we expose the right information via the `DotLottieFile` info
 2. while this may be useful occasionally, I'm not sure it's an operation that would be used enough to warrant being included as public API

What would you think about removing these methods, in favor of creating public APIs on `DotLottieFile` that let you implement this yourself? (e.g. public APIs for getting the list of animations in the file).

Reply:
I agree with you, they don't make sense to most people. I'll change the PR to make necessary functions public and remove those methods